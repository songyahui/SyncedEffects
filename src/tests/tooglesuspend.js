"use hopscript"

const hh = require("hiphop");

const mach = new hh.ReactiveMachine(
   <hh.module toogle>
     <hh.suspend toggleSignal=toogle>
       <hh.loop>
	 <hh.atom apply=${function() {
	    console.log("plop");
	 }}/>
	 <hh.pause/>
       </hh.loop>
     </hh.suspend>
   </hh.module>
);

mach.debug_emitted_func = console.log;

mach.react();
mach.react();
mach.inputAndReact('toogle');
mach.react();
mach.react();
mach.inputAndReact('toogle');
mach.react();
mach.react();
