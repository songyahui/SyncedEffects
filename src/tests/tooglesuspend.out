plop
[]
plop
[]
[ 'toogle' ]
[]
[]
plop
[ 'toogle' ]
plop
[]
plop
[]
